[2025-10-08T04:41:07.358+0000] {local_task_job_runner.py:120} INFO - ::group::Pre task execution logs
[2025-10-08T04:41:07.443+0000] {taskinstance.py:2076} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: restaurant_analytics.run_spark_stream manual__2025-10-08T04:35:34.590264+00:00 [queued]>
[2025-10-08T04:41:07.465+0000] {taskinstance.py:2076} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: restaurant_analytics.run_spark_stream manual__2025-10-08T04:35:34.590264+00:00 [queued]>
[2025-10-08T04:41:07.467+0000] {taskinstance.py:2306} INFO - Starting attempt 2 of 2
[2025-10-08T04:41:07.516+0000] {taskinstance.py:2330} INFO - Executing <Task(BashOperator): run_spark_stream> on 2025-10-08 04:35:34.590264+00:00
[2025-10-08T04:41:07.535+0000] {warnings.py:112} WARNING - /home/***/.local/lib/python3.12/site-packages/***/task/task_runner/standard_task_runner.py:62: DeprecationWarning: This process (pid=426) is multi-threaded, use of fork() may lead to deadlocks in the child.
  pid = os.fork()

[2025-10-08T04:41:07.540+0000] {standard_task_runner.py:64} INFO - Started process 430 to run task
[2025-10-08T04:41:07.539+0000] {standard_task_runner.py:90} INFO - Running: ['***', 'tasks', 'run', 'restaurant_analytics', 'run_spark_stream', 'manual__2025-10-08T04:35:34.590264+00:00', '--job-id', '131', '--raw', '--subdir', 'DAGS_FOLDER/restaurant_dag.py', '--cfg-path', '/tmp/tmpkih883w0']
[2025-10-08T04:41:07.545+0000] {standard_task_runner.py:91} INFO - Job 131: Subtask run_spark_stream
[2025-10-08T04:41:07.705+0000] {task_command.py:426} INFO - Running <TaskInstance: restaurant_analytics.run_spark_stream manual__2025-10-08T04:35:34.590264+00:00 [running]> on host c34aac175a74
[2025-10-08T04:41:07.874+0000] {taskinstance.py:2648} INFO - Exporting env vars: AIRFLOW_CTX_DAG_EMAIL='quantruong1918@gmail.com' AIRFLOW_CTX_DAG_OWNER='***' AIRFLOW_CTX_DAG_ID='restaurant_analytics' AIRFLOW_CTX_TASK_ID='run_spark_stream' AIRFLOW_CTX_EXECUTION_DATE='2025-10-08T04:35:34.590264+00:00' AIRFLOW_CTX_TRY_NUMBER='2' AIRFLOW_CTX_DAG_RUN_ID='manual__2025-10-08T04:35:34.590264+00:00'
[2025-10-08T04:41:07.877+0000] {taskinstance.py:430} INFO - ::endgroup::
[2025-10-08T04:41:07.879+0000] {subprocess.py:63} INFO - Tmp dir root location: /tmp
[2025-10-08T04:41:07.881+0000] {subprocess.py:75} INFO - Running command: ['/usr/bin/bash', '-c', '/opt/spark/bin/spark-submit --master local[*] /opt/***/dags/spark_streaming.py']
[2025-10-08T04:41:07.909+0000] {subprocess.py:86} INFO - Output:
[2025-10-08T04:41:14.737+0000] {subprocess.py:93} INFO - Traceback (most recent call last):
[2025-10-08T04:41:14.738+0000] {subprocess.py:93} INFO -   File "/opt/***/dags/spark_streaming.py", line 5, in <module>
[2025-10-08T04:41:14.742+0000] {subprocess.py:93} INFO -     from pyspark.ml import PipelineModel
[2025-10-08T04:41:14.742+0000] {subprocess.py:93} INFO -   File "/opt/spark/python/lib/pyspark.zip/pyspark/ml/__init__.py", line 22, in <module>
[2025-10-08T04:41:14.743+0000] {subprocess.py:93} INFO -   File "/opt/spark/python/lib/pyspark.zip/pyspark/ml/base.py", line 40, in <module>
[2025-10-08T04:41:14.744+0000] {subprocess.py:93} INFO -   File "/opt/spark/python/lib/pyspark.zip/pyspark/ml/param/__init__.py", line 32, in <module>
[2025-10-08T04:41:14.745+0000] {subprocess.py:93} INFO - ModuleNotFoundError: No module named 'numpy'
[2025-10-08T04:41:14.807+0000] {subprocess.py:93} INFO - 25/10/08 04:41:14 INFO ShutdownHookManager: Shutdown hook called
[2025-10-08T04:41:14.814+0000] {subprocess.py:93} INFO - 25/10/08 04:41:14 INFO ShutdownHookManager: Deleting directory /tmp/spark-d8da204d-c36a-4271-8531-91a72f383ced
[2025-10-08T04:41:14.878+0000] {subprocess.py:97} INFO - Command exited with return code 1
[2025-10-08T04:41:14.882+0000] {taskinstance.py:441} INFO - ::group::Post task execution logs
[2025-10-08T04:41:14.908+0000] {taskinstance.py:2905} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 465, in _execute_task
    result = _execute_callable(context=context, **execute_callable_kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 432, in _execute_callable
    return execute_callable(context=context, **execute_callable_kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/baseoperator.py", line 401, in wrapper
    return func(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/operators/bash.py", line 243, in execute
    raise AirflowException(
airflow.exceptions.AirflowException: Bash command failed. The command returned a non-zero exit code 1.
[2025-10-08T04:41:14.921+0000] {taskinstance.py:1206} INFO - Marking task as FAILED. dag_id=restaurant_analytics, task_id=run_spark_stream, run_id=manual__2025-10-08T04:35:34.590264+00:00, execution_date=20251008T043534, start_date=20251008T044107, end_date=20251008T044114
[2025-10-08T04:41:14.945+0000] {standard_task_runner.py:110} ERROR - Failed to execute job 131 for task run_spark_stream (Bash command failed. The command returned a non-zero exit code 1.; 430)
[2025-10-08T04:41:14.995+0000] {local_task_job_runner.py:243} INFO - Task exited with return code 1
[2025-10-08T04:41:15.019+0000] {taskinstance.py:3503} INFO - 0 downstream tasks scheduled from follow-on schedule check
[2025-10-08T04:41:15.031+0000] {local_task_job_runner.py:222} INFO - ::endgroup::
